File: packages/firebase_in_app_messaging/android/src/main/java/com/example/firebase_in_app_messaging/FirebaseInAppMessagingPlugin.java
package com.example.firebase_in_app_messaging;

import com.google.firebase.inappmessaging.FirebaseInAppMessaging;
import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;
import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
import io.flutter.plugin.common.MethodChannel.Result;
import io.flutter.plugin.common.PluginRegistry.Registrar;

/** FirebaseInAppMessagingPlugin */
public class FirebaseInAppMessagingPlugin implements MethodCallHandler {
  private final FirebaseInAppMessaging instance;

  public static void registerWith(Registrar registrar) {
    final MethodChannel channel =
        new MethodChannel(registrar.messenger(), "plugins.flutter.io/firebase_in_app_messaging");
    channel.setMethodCallHandler(new FirebaseInAppMessagingPlugin());
  }

  private FirebaseInAppMessagingPlugin() {
    instance = FirebaseInAppMessaging.getInstance();
  }

  @Override
  public void onMethodCall(MethodCall call, Result result) {
    switch (call.method) {
      case "triggerEvent":
        {
          String eventName = call.argument("eventName");
          instance.triggerEvent(eventName);
          result.success(null);
          break;
        }
      case "setMessagesSuppressed":
        {
          Boolean suppress = (Boolean) call.arguments;
          instance.setMessagesSuppressed(suppress);
          result.success(null);
          break;
        }
      case "setAutomaticDataCollectionEnabled":
        {
          Boolean enabled = (Boolean) call.arguments;
          instance.setAutomaticDataCollectionEnabled(enabled);
          result.success(null);
          break;
        }
      default:
        {
          result.notImplemented();
          break;
        }
    }
  }
}

File: packages/firebase_in_app_messaging/ios/Classes/FirebaseInAppMessagingPlugin.m
// Copyright 2019 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#import "FirebaseInAppMessagingPlugin.h"

#import <Firebase/Firebase.h>

@implementation FirebaseInAppMessagingPlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar {
  FlutterMethodChannel *channel =
      [FlutterMethodChannel methodChannelWithName:@"plugins.flutter.io/firebase_in_app_messaging"
                                  binaryMessenger:[registrar messenger]];
  FirebaseInAppMessagingPlugin *instance = [[FirebaseInAppMessagingPlugin alloc] init];
  [registrar addMethodCallDelegate:instance channel:channel];
}

- (instancetype)init {
  self = [super init];
  if (self) {
    if (![FIRApp appNamed:@"__FIRAPP_DEFAULT"]) {
      NSLog(@"Configuring the default Firebase app...");
      [FIRApp configure];
      NSLog(@"Configured the default Firebase app %@.", [FIRApp defaultApp].name);
    }
  }
  return self;
}

- (void)handleMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result {
  if ([@"triggerEvent" isEqualToString:call.method]) {
    NSString *eventName = call.arguments[@"eventName"];
    FIRInAppMessaging *fiam = [FIRInAppMessaging inAppMessaging];
    [fiam triggerEvent:eventName];
    result(nil);
  } else if ([@"setMessagesSuppressed" isEqualToString:call.method]) {
    NSNumber *suppress = [NSNumber numberWithBool:call.arguments];
    FIRInAppMessaging *fiam = [FIRInAppMessaging inAppMessaging];
    fiam.messageDisplaySuppressed = [suppress boolValue];
    result(nil);
  } else if ([@"setAutomaticDataCollectionEnabled" isEqualToString:call.method]) {
    NSNumber *enabled = [NSNumber numberWithBool:call.arguments];
    FIRInAppMessaging *fiam = [FIRInAppMessaging inAppMessaging];
    fiam.automaticDataCollectionEnabled = [enabled boolValue];
    result(nil);
  } else {
    result(FlutterMethodNotImplemented);
  }
}

@end
